pkg/syncer/spec/spec_process.go

```diff
@@ -351,10 +352,10 @@ func transformName(syncedObject *unstructured.Unstructured) {
 	if syncedObject.GroupVersionKind() == configMapGVR && syncedObject.GetName() == "kube-root-ca.crt" {
 		syncedObject.SetName("kcp-root-ca.crt")
 	}
-	// TODO(jmprusi): We are rewriting the name of the object into a non random one so we can reference it from the deployment transformer
-	//                but this means that means than more than one default-token-XXXX object will overwrite the same "kcp-default-token"
-	//				  object. This must be fixed.
-	if syncedObject.GroupVersionKind() == secretGVR && strings.Contains(syncedObject.GetName(), "default-token-") {
-		syncedObject.SetName("kcp-default-token")
+	if syncedObject.GroupVersionKind() == secretGVR && strings.Contains(syncedObject.GetName(), "-token-") {
+		// Only transform the name if the object is service-account token.
+		if _, ok := syncedObject.GetAnnotations()[core.ServiceAccountNameKey]; ok {
+			syncedObject.SetName("kcp-" + syncedObject.GetName())
```

Comment:
Could you please remind me why we need to change the name of the token secret in the pcluster?

Reply:
Ok, that's something we should discuss:

As you know, KCP and Kubernetes create a "default" service account on each namespace (this is going to change in the next k8s release afaik), and that service account has a token created like "default-token-XXXX" where "XXXX" is a random value.

So, what I do here is to add the "kcp-" suffix to the name of the token, to make it easier to be distinguished from the pcluster default tokens, and also, eliminating the possible "conflict" of names (perhaps, this is slightly farfetched).

One possibility would be to just rename the default token, and don't rename any other token created by the user on kcp, but this means controlling for the "special" case (default-token-XXXX)... 

The other idea is just, to stop renaming any token, and hope🤞🏼 we don't have conflicts downstream.

Reply:
I think there is a risk that we could have a conflict (at least as long as automatic token secret creation is enabled, which is pre Kubernetes 1.24), so it's probably best to do some renaming.

We also need to make sure that the renamed name does not exceed the maximum allowed name length for a secret, which follows the DNS subdomain rules (https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names).

The most foolproof thing is probably to try to prepend `kcp-`, but if that exceeds the maximum allowed length, then do `kcp-<original value minus 8 characters>-<7 character hash of the original name>` or something along those lines. WDYT @sttts @jmprusi?

Also, could you please create a helper function to do the name transformation, and use it both here and in the deployment mutator?

Reply:
Makes sense, on it.

Reply:
done! please take a look at it